src/luarocks/util.lua

```diff
@@ -412,9 +412,6 @@ end
 
 do
    local function Q(pathname)
-      if pathname:match("^.:") then
-         return pathname:sub(1, 2) .. '"' .. pathname:sub(3) .. '"'
-      end
```

Comment:
this looks weird indeed, seems to quote a path, except for the drive letter, which is kept outside the quotes.

fwiw, was added here: https://github.com/luarocks/luarocks/commit/44f363fe3dcf60540b394f798f894c1ed2b26ebc#diff-947edb02b7ef84720ff6067cb1375fda3d61546ecd07f1de756e184e47f67f2cR47-R51

maybe @hishamhm remembers why it was added.

Reply:
An excerpt from the failed AppVeyor build:
```
stack traceback:
	.//spec\util\test_env.lua:678: in function 'build_environment'
	.//spec\util\test_env.lua:1125: in function 'main'
	.//spec\util\test_env.lua:1146: in function 'setup_specs'
	spec\quick_spec.lua:7: in function <spec\quick_spec.lua:6>
```
A couple of points:
- The error was thrown while setting up the tesing environment, which shouldn't be affected by my change.
- `spec/util/test_env.lua` *only has 950 lines, so how is there an error at line 1146?*